lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -1058,3 +1058,36 @@ void PartialConnectOp::getCanonicalizationPatterns(RewritePatternSet &results,
                                                    MLIRContext *context) {
   addCanonicalizerMethod<PartialConnectOp>(results, context);
 }
+
+LogicalResult MemOp::canonicalize(MemOp op, PatternRewriter &rewriter) {
+  // If memory has known, but zero width, eliminate it.  LowerTypes is run
+  // before this, so we can't leave structs sitting around.  Thus we must do the
+  // recursion to fix up the ports.
```

Comment:
I'm a bit confused about this comment.  Canonicalization is a pass that can be run at any time - both before and after lower types.  Firtool is one set of policies, but the pass should work anywhere.  I think it is safe, just confused about the comment.